softserve/views.py

```diff
@@ -84,21 +85,25 @@ def get_node_data():
         counts = request.form['counts']
         name = request.form['node_name']
         hours_ = request.form['hours']
-        pubkey_ = request.form['pubkey']
 
         # Validating the hours and node counts
         if int(counts) > 5 or int(hours_) > 4 or int(counts) > n:
             flash('Please enter the valid data')
             logging.exception('User entered the invalid hours or counts value')
             return render_template('form.html', n=n)
 
+        # retreive key from github account
+        key = subprocess.Popen(['curl', 'https://github.com/%s.keys' % g.user.username],
```

Comment:
Use `requests` to pull this from github rather than a Popen. Can we cache the key in the user account so we don't do a curl every time?

Reply:
I thought of this but there could be a possibility of user deleting or updating the key on github account. With that, there needs to be proper recheck of cached keys and so on updating the user's keypair created on Rackspace.

Reply:
How often does a person cycle their key? Once in a while, right? We could create a different method to pull in the latest key from Github.